zerver/actions/realm_settings.py

```diff
@@ -486,3 +487,57 @@ def do_send_realm_reactivation_email(realm: Realm, *, acting_user: Optional[User
         language=language,
         context=context,
     )
+
+
+def do_send_realm_deactivation_email(realm: Realm, acting_user: Optional[UserProfile]) -> None:
+    # if realm is not deactivated, but changed the domain
+    if not acting_user:
+        return
+
+    event_time = timezone_now()
+    RealmAuditLog.objects.create(
+        realm=realm,
+        acting_user=acting_user,
+        event_type=RealmAuditLog.REALM_DEACTIVATION_EMAIL_SENT,
+        event_time=event_time,
+        extra_data=orjson.dumps(
+            {
+                RealmAuditLog.ROLE_COUNT: realm_user_count_by_role(realm),
+            }
+        ).decode(),
+    )
+    context = {
+        "realm_uri": realm.uri,
+        "realm_name": realm.name,
+        "event_date": event_time.date(),
+        "acting_user_name": acting_user.full_name,
+        "send_to_deactivating_owner": True,
+    }
+    language = realm.default_language
+    owners = realm.get_human_owner_users()
+
+    deactivating_owner = acting_user
+    other_owners = owners.exclude(id=deactivating_owner.id)
+
+    # Send separate emails to the deactivating owner and the rest of the owners
+    send_email(
+        "zerver/emails/realm_deactivated",
+        to_emails=[deactivating_owner.delivery_email],
+        from_name=FromAddress.security_email_from_name(language=language),
+        from_address=FromAddress.tokenized_no_reply_address(),
+        language=language,
+        context=context,
+        realm=realm,
+    )
```

Comment:
This block needs to of course be conditional on `deactivating_owner` being not None?